graylog2-web-interface/src/components/grok-patterns/GrokPatternInput.jsx

```diff
@@ -0,0 +1,137 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { Input } from 'components/bootstrap';
+import { Row, Col, Button, ListGroup, ListGroupItem } from 'react-bootstrap';
+
+import GrokPatternInputStyle from './GrokPatternInput.css';
+
+class GrokPatternInput extends React.Component {
+  static propTypes = {
+    pattern: PropTypes.string,
+    patterns: PropTypes.array,
+    onPatternChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    pattern: '',
+    patterns: [],
+    onPatternChange: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      patternFilter: '',
+      activeListItem: -1,
+    };
+  }
+
+  shownListItems = [];
+
+  _onPatternChange = (e) => {
+    this.props.onPatternChange(e.target.value);
+  };
+
+  _onPatternFilterChange = (e) => {
+    this.setState({ patternFilter: e.target.value, activeListItem: -1 });
+  };
+
+  _onPatternFilterKeyDown = (e) => {
+    const ARROW_DOWN = 40;
+    const ARROW_UP = 38;
+    const ENTER = 13;
+    const listItem = this.shownListItems[this.state.activeListItem];
+
+    let activeListItem = 0;
+    let domElement;
+    switch (e.keyCode) {
```

Comment:
Moving down with the keyboard has issues and sometimes scrolls the outer component, in this case the modal:

![scroll](https://user-images.githubusercontent.com/716185/41916502-76afc814-7958-11e8-8bae-fbcea836ce21.gif)